torch/sparse/_triton_ops.py

```diff
@@ -835,7 +844,50 @@ def kernel(grid, *sliced_tensors):
             size=input.shape,
             layout=input.layout
         )
+
+    def _scaled_dot_product_attention(
+        query: torch.Tensor,
+        key: torch.Tensor,
+        value: torch.Tensor,
+        attn_mask: Optional[torch.Tensor],
+        dropout_p: float = 0.0,
+        is_causal: bool = False
+    ):
+        f_name = "_scaled_dot_product_attention"
+        check(
+            not is_causal,
+            f"{f_name}(): is_causal == True is not supported."
+        )
+        check(
+            attn_mask is not None,
+            f"{f_name}(): attn_mask == None is not supported."
+        )
+        assert attn_mask is not None
+
+        check(
+            attn_mask.layout == torch.sparse_bsr,
```

Comment:
Are float masks supported just not tested?

Reply:
They are tested.

Reply:
I am new to this test structure, maybe I missed out

Reply:
There is a `mask_dtype` argument in the test. This floating mask support is similar to the boolean case, i.e. there is no mapping to `{0, -inf}` as in the strided case.

Reply:
you are right the formatting was weird on my monitor and thought it was a different line 👍